verilog/tools/ls/autoexpand.cc

```diff
@@ -1402,25 +1368,77 @@ std::optional<absl::string_view> AutoExpander::FindSpanToReplace(
   return replaced_span;
 }
 
-// Converts AutoExpander::Expansion structs to LSP TextEdits
-std::vector<TextEdit> ConvertAutoExpansionsToTextEdits(
+// Converts AutoExpander::Expansion structs to LSP TextEdits and performs
+// formatting on them if possible
+std::vector<TextEdit> ConvertAutoExpansionsToFormattedTextEdits(
     const TextStructureView &text_structure,
-    const std::vector<AutoExpander::Expansion> &expansions) {
-  std::vector<TextEdit> edits;
-  edits.reserve(expansions.size());
+    std::vector<AutoExpander::Expansion> expansions) {
+  std::sort(expansions.begin(), expansions.end(),
+            [](const AutoExpander::Expansion &first,
+               const AutoExpander::Expansion &second) {
+              return first.replaced_span.begin() < second.replaced_span.begin();
+            });
+
+  // To format expansions, we need to apply them to the source. Then we only
+  // call the formatter once, and produce a single TextEdit that replaces the
+  // entire file. This is orders of magnitude faster than formatting individual
```

Comment:
Maybe clarify in the comment, that you still limit the formatter ranges to the affected line ranges generated by the auto expander. Otherwise this comment sounds like the user ends up with the full file reformatted, but all we want is the snippets that have been auto-expand modified.